src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -2244,84 +2126,58 @@ namespace HVACUnitaryBypassVAV {
-        InletNode = CBVAV(CBVAVNum).AirInNode;
-        MixerOutsideAirNode = CBVAV(CBVAVNum).MixerOutsideAirNode;
-        MixerReliefAirNode = CBVAV(CBVAVNum).MixerReliefAirNode;
+        int OutletNode = CBVAV(CBVAVNum).AirOutNode;
+        int InletNode = CBVAV(CBVAVNum).AirInNode;
+        int MixerOutsideAirNode = CBVAV(CBVAVNum).MixerOutsideAirNode;
+        int MixerReliefAirNode = CBVAV(CBVAVNum).MixerReliefAirNode;
         if (CBVAV(CBVAVNum).CondenserNodeNum > 0) {
-            OutdoorDryBulbTemp = Node(CBVAV(CBVAVNum).CondenserNodeNum).Temp;
-            OutdoorBaroPress = Node(CBVAV(CBVAVNum).CondenserNodeNum).Press;
+            OutdoorDryBulbTemp = DataLoopNode::Node(CBVAV(CBVAVNum).CondenserNodeNum).Temp;
+            OutdoorBaroPress = DataLoopNode::Node(CBVAV(CBVAVNum).CondenserNodeNum).Press;
         } else {
-            OutdoorDryBulbTemp = OutDryBulbTemp;
-            OutdoorBaroPress = OutBaroPress;
+            OutdoorDryBulbTemp = DataEnvironment::OutDryBulbTemp;
+            OutdoorBaroPress = DataEnvironment::OutBaroPress;
         }
 
         SaveCompressorPLR = 0.0;
 
         // Bypass excess system air through bypass duct and calculate new mixed air conditions at OA mixer inlet node
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp =
-            (1.0 - BypassDuctFlowFraction) * Node(InletNode).Temp + BypassDuctFlowFraction * Node(OutletNode).Temp;
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat =
-            (1.0 - BypassDuctFlowFraction) * Node(InletNode).HumRat + BypassDuctFlowFraction * Node(OutletNode).HumRat;
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).Enthalpy =
-            PsyHFnTdbW(Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp, Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat);
-        SimOAMixer(CBVAV(CBVAVNum).OAMixName, FirstHVACIteration, CBVAV(CBVAVNum).OAMixIndex);
-
-        if (CBVAV(CBVAVNum).FanPlace == BlowThru) {
+        if (CBVAV(CBVAVNum).plenumIndex > 0 || CBVAV(CBVAVNum).mixerIndex > 0) {
+            DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode) = DataLoopNode::Node(InletNode);
+        } else {
+            DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp =
+                (1.0 - BypassDuctFlowFraction) * DataLoopNode::Node(InletNode).Temp + BypassDuctFlowFraction * DataLoopNode::Node(OutletNode).Temp;
+            DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat = (1.0 - BypassDuctFlowFraction) * DataLoopNode::Node(InletNode).HumRat +
+                                                                           BypassDuctFlowFraction * DataLoopNode::Node(OutletNode).HumRat;
+            DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).Enthalpy = Psychrometrics::PsyHFnTdbW(
+                DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp, DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat);
+        }
+        MixedAir::SimOAMixer(CBVAV(CBVAVNum).OAMixName, FirstHVACIteration, CBVAV(CBVAVNum).OAMixIndex);
```

Comment:
RE: don't need to call mixer when internal OA flow = 0 (i.e., when user sets internal OA flow = 0 or uses AirloopHVAC:Mixer or ReturnPlenum and sets internal OA flow = 0). I decided to leave this alone. I could have done this but thought it was overkill (but might be a time saver overall):

```
if (OAFlow == 0.0) {
  Node(MixedAirNode) = Node(MixerInletNode);
  Node(MixerExhNode) = Node(MixerInletNode);
  Node(MixerExhNode).MassFlowRate = 0.0;
} else{
  MixedAir::SimOAMixer();
}
```